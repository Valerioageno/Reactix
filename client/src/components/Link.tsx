import React, { useState } from 'react'

interface LinkProps {
   link: string
   children: string
}

export default function Link(props: LinkProps): JSX.Element {
   const [hovered, setHovered] = useState<boolean>(false)

   return (
      <a
         className={hovered ? 'link linkHovered' : 'link'} //Just for example testing purpose...
         target='_blank'
         onMouseEnter={() => setHovered(true)}
         onMouseLeave={() => setHovered(false)}
         href={props.link || '#'}
         rel='noreferrer'
      >
         {props.children}
      </a>
   )
}
