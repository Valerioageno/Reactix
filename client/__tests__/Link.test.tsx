import React from 'react'
import renderer from 'react-test-renderer'
import Link from '../src/components/Link'

test('Link change class when hovered', () => {
   const component = renderer.create(
      <Link link='https://github.com/Valerioageno/reactix'>Repo</Link>
   )

   let tree = component.toJSON()
   expect(tree).toMatchSnapshot()

   // manually trigger the callback
   renderer.act(() => tree?.props.onMouseEnter())
   // re-rendering
   tree = component.toJSON()
   expect(tree).toMatchSnapshot()

   renderer.act(() => tree?.props.onMouseLeave())
   // re-rendering
   tree = component.toJSON()
   expect(tree).toMatchSnapshot()
})
