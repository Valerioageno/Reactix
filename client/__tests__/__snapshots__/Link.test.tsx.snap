// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link change class when hovered 1`] = `
<a
  className="link"
  href="https://github.com/Valerioageno/reactix"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  rel="noreferrer"
  target="_blank"
>
  Repo
</a>
`;

exports[`Link change class when hovered 2`] = `
<a
  className="link linkHovered"
  href="https://github.com/Valerioageno/reactix"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  rel="noreferrer"
  target="_blank"
>
  Repo
</a>
`;

exports[`Link change class when hovered 3`] = `
<a
  className="link"
  href="https://github.com/Valerioageno/reactix"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  rel="noreferrer"
  target="_blank"
>
  Repo
</a>
`;
